cap/modules/deposit/serializers/schemas/json.py

```diff
@@ -51,34 +50,28 @@ class DepositFormSchema(DepositSchema):
     """Schema for deposit v1 in JSON."""
 
     schemas = fields.Method('get_deposit_schemas', dump_only=True)
-
     repositories = fields.Method('get_repositories', dump_only=True)
     workflows = fields.Method('get_workflows', dump_only=True)
 
     can_update = fields.Method('can_user_update', dump_only=True)
     can_admin = fields.Method('can_user_admin', dump_only=True)
 
     def get_repositories(self, obj):
-        deposit = CAPDeposit.get_record(obj['pid'].object_uuid)
-        webhooks = deposit.model.webhooks
-
-        return DepositRepositoriesSchema(many=True).dump(webhooks).data
+        hooks = obj['deposit'].model.webhooks
+        return DepositRepositoriesSchema(many=True).dump(hooks).data
 
     def get_workflows(self, obj):
-        deposit = CAPDeposit.get_record(obj['pid'].object_uuid)
-        workflows = deposit.model.reana_workflows
-
+        workflows = obj['deposit'].model.workflows
         return ReanaWorkflowSchema(many=True).dump(workflows).data
 
     def get_deposit_schemas(self, obj):
-        deposit = CAPDeposit.get_record(obj['pid'].object_uuid)
-
-        schema = current_jsonschemas.get_schema(deposit.schema.deposit_path,
-                                                with_refs=True,
-                                                resolved=True)
-        uiSchema = deposit.schema.deposit_options
-
-        return dict(schema=copy.deepcopy(schema), uiSchema=uiSchema)
+        ui_schema = obj['deposit'].schema.deposit_options
+        schema = current_jsonschemas.get_schema(
+            obj.schema.deposit_path,
```

Comment:
This doesn't work because `obj` (dictionary passed to your serializer) doesn't have a schema object. 
You want to access obj['deposit'], which is assigned to `CAPDeposit` object - this one has a schema property.
```suggestion
            obj['deposit'].schema.deposit_path,
```

Reply:
resolved:)